### PR TITLE
Config: Port CPU overclock setting to new config.

### PR DESCRIPTION
**Pull request by AdmiralCurtiss**
Mostly so I can get a review on if this is the correct way to do it to avoid losing perf in hot code, in particular the config change callback.

### CODE REVIEW

File: Source/Core/Core/CoreTiming.cpp [hunk 1]
```diff
@@ -148,6 +151,18 @@ void Shutdown()
...
+void RefreshConfig()
+{
+  if (!s_has_registered_config_callback)
+  {
+    Config::AddConfigChangedCallback([]() { Core::RunAsCPUThread([]() { RefreshConfig(); }); });
+    s_has_registered_config_callback = true;
+  }
+
+  s_config_OC_factor =
+      Config::Get(Config::MAIN_OVERCLOCK_ENABLE) ? Config::Get(Config::MAIN_OVERCLOCK) : 1.0f;
```
Comment (JosJuice) [inline Source/Core/Core/CoreTiming.cpp:163]: Perhaps consider also having an `s_config_OC_factor_inverted` to let us skip a division in `Advance`. (Maybe not all that important, but hey, Advance does run fairly often...)

**Review by JosJuice — COMMENTED** [review]
The implementation seems reasonable. But have we tested that there actually is any noticeable performance loss when doing this without the second commit?

**Comment by AdmiralCurtiss** [comment]
re perf, as always this is very unscientific but idling in the Melee main menu I get roughly:

before: ~390 FPS
without last commit: ~370 FPS
with last commit: ~390 FPS

**Comment by JosJuice** [comment]
Alright, it definitely sounds like we should go with the caching then, considering it's a fairly simple thing to add.

**Review by JosJuice — APPROVED** [review]

**Pull request merged**